src/rez/wheel.py

```diff
@@ -0,0 +1,570 @@
+            variants["platform"] = platform_name()
+
+        if key == "tag":
+            # May occur multiple times
+            #
+            # Example:
+            #   py2-none-any
+            #   py3-none-any
+            #   cp36-cp36m-win_amd64
+            #
+            py_tag, abi_tag, plat_tag = value.split("-")
+            major_ver = py_tag[2]
+
+            py[major_ver] = True
+
+            if plat_tag != "any":
+                # We could convert e.g. `win_amd64` to a Rez platform
+                # and os version, such as `platform-windows` and
+                # `os-windows.10.0.1800` but it's safe to assume that if
+                # this package was provided by pip, it must be specific
+                # to the currently running platform and os.
+
+                variants["os"] = os_name()
+                variants["platform"] = platform_name()  # e.g. windows
+
+                # Indicate that this week depends on the Python minor version
+                # which is true of any compiled Python package.
+                py["minor"] = True
+
+    if py["minor"]:
+        # Use the actual version from the running Python
+        # rather than what's coming out of the the WHEEL
+        # See 
+        variants["python"] = python_version()
+
+    elif py["2"] and py["3"]:
+        variants["python"] = None
+
+    elif py["2"]:
+        variants["python"] = "2"
+
+    elif py["3"]:
+        variants["python"] = "3"
+
+    return [
+        k + "-" + variants[k]
+
+        # Order is important
+        for k in ("platform",
+                  "os",
+                  "python")
+
+        if variants[k] is not None
+    ]
+
+
+def os_name():
+    """Return pip-compatible OS, e.g. windows-10.0 and Debian-7.6"""
+    # pip packages are no more specific than minor/major of an os
+    # E.g. windows-10.0.18362 -> windows-10.0
+    return ".".join(platform_.os.split(".")[:2])
+
+
+def platform_name():
+    return platform_.name
+
+
+@lru_cache()
+def python_version():
+    """Return major.minor version of Python, prefer current context"""
+
+    import subprocess
+    from rez.status import status
+    context = status.context
+
+    try:
+        # Use supplied Python
+        package = context.get_resolved_package("python")
+        return ".".join(str(v) for v in package.version[:2])
+
+    except AttributeError:
+        # In a context, but no Python was found
+        pass
+
+    except IndexError:
+        # We'll need this for almost every package on PyPI
+        raise IndexError("%s didn't have a minor version" % package.uri)
+
+    # Try system Python
+    popen = subprocess.Popen(
+        """\
+        python -c "import sys;print('.'.join(map(str, sys.version_info[:2])))"
+        """,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT,
+        universal_newlines=True,
+        bufsize=10 ** 4,  # Enough to capture the version
+        shell=True,
```

Comment:
Same here, any reason to use `shell=True`?

Reply:
Yes, in both cases, `shell=True` enables `rez wheel` to be prefixed with `rez env -- rez wheel`. Without it, it wouldn't be able to see the Python exposed by `rez env`.